frontend/src/pages/secret-manager/OverviewPage/components/SecretOverviewTableRow/SecretEditRow.tsx

```diff
@@ -280,30 +234,53 @@ export const SecretEditRow = ({
     setIsModalOpen(false);
 
     try {
-      await onSecretDelete(environment, secretName, secretId);
+      await onSecretDelete(environment, secretName, SecretType.Shared);
       reset({ value: null });
     } finally {
       setIsDeleting.off();
     }
   }, [onSecretDelete, environment, secretName, secretId, reset, setIsDeleting]);
 
+  const openCreateSharedSecretPopup = useCreateSharedSecretPopup({
+    getFetchedValue: () => getValues("value") || undefined,
+    fetchSecretParams: fetchSecretValueParams
+  });
+
+  const deleteLabel = useMemo(() => {
+    if (isRotatedSecret) return "Cannot Delete Rotated Secret";
+    return "Delete";
+  }, []);
+
+  const createPersonalSecretOverride = useCreatePersonalSecretOverride(currentProject.id);
+
+  const handleAddPersonalOverride = useCallback(async () => {
+    createPersonalSecretOverride(
+      {
+        key: secretName,
+        environment,
+        secretPath
+      },
+      secretValueData?.value
+    );
+  }, [createPersonalSecretOverride, secretValueData?.value]);
```

Comment:
This logic bring over the logic to share a secret and create a personal override to the overview page.